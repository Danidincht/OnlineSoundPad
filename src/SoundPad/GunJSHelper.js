import Gun from 'gun';

export function getGunInstance() {
	return Gun({
		peers: 'https://gun-manhattan.herokuapp.com/gun',
		localStorage: false
	});
}

export function getItemsNode(roomName) {
	return getGunInstance().get(roomName)
		.get('soundPad')
		.get('items');
}