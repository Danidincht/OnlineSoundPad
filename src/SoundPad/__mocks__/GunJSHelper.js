const gunHelper = jest.createMockFromModule('../GunJSHelper');
module.exports = gunHelper;