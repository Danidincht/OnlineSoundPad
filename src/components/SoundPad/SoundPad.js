import React from 'react';
import ItemInput from '../ItemInput/ItemInput';
import ItemList from '../ItemList/ItemList';
import Gun from 'gun';
import 'gun/lib/radix';
import 'gun/lib/radisk';
import 'gun/lib/store';
import 'gun/lib/rindexed';

function SoundPad() {
	var opt = {
		peers: 'https://gun-manhattan.herokuapp.com/gun',
		localStorage: false
	};

	var gun = Gun(opt);
	var soundPadNode = roomNode.get('soundPad');
	var itemsNode = soundPadNode.get('items');

	return (
	<div>
		<ItemInput itemsNode={itemsNode}></ItemInput>
		<ItemList itemsNode={itemsNode}></ItemList>
	</div>
)};

export default SoundPad;