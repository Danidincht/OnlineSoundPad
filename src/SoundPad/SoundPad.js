import './SoundPad.css';
import PadItem from '../PadItem/PadItem';

function SoundPad() {
	const items = [
		createPadItemContent('PadItem1', 'https://www.sousound.com/music/healing/healing_01.mp3'),
		createPadItemContent('PadItem2', 'https://www.sousound.com/music/healing/healing_02.mp3')
	];
	return (
		<div>
			Sound Pad Online
			{
				items.map((item) => (
					<PadItem 
						title={item.title}
						audio={item.audio}
					/>
				))
			}
		</div>
	);
}

function createPadItemContent(title, audio) {
	return {
		title,
		audio
	}
}

export default SoundPad;
