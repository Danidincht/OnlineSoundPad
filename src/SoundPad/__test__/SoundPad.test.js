jest.mock('../GunJSHelper');

import React, { useState as useStateMock } from 'react';
import { render } from '@testing-library/react';
import SoundPad from '../SoundPad';
import { getItemsNode } from '../GunJSHelper';

jest.mock('react', () => ({
	...jest.requireActual('react'),
	useState: jest.fn()
}));
const setItemMapMock = jest.fn();

jest.mock('#c/PadEditor', () => 'PadEditorMock');
jest.mock('#c/PadItem', () => 'PadItemMock');

beforeEach(() => {
	useStateMock.mockImplementation((init) => [init, setItemMapMock]);
	getItemsNode.mockReturnValue({
		map: () => ({
			on: jest.fn()
		})
	});
});

afterEach(jest.clearAllMocks);

it('renders basic Sound Pad layout', () => {
	// Given - When
	const page = render(<SoundPad />).baseElement;

	// Then
	expect(page).toMatchSnapshot();
});

it('renders items from GunJS', () => {
	// Given
	var items = new Map([
		[1, {text: 'text1', audio: {}}],
		[2, {text: 'text2', audio: {}}]
	]);
	useStateMock.mockImplementation(() => [items, setItemMapMock]);

	// When
	const page = render(<SoundPad />).baseElement;

	// Then
	expect(page).toMatchSnapshot();
});

describe('itemsNode on function callback', () => {
	let gunJSOnCallback;
	beforeEach(() => {
		gunJSOnCallback = undefined;
		getItemsNode.mockReturnValue({
			map: () => ({
				on: (cb) => {
					gunJSOnCallback = cb;
				}
			})
		});
	});

	it('Adds items to itemMap to render if data is not null', () => {
		// Given
		render(<SoundPad />).baseElement;

		// When
		gunJSOnCallback('non-null object');

		// Then
		expect(setItemMapMock).toBeCalledTimes(1);
	});

	it('Does not add item to itemMap to render if data is null', () => {
		// Given
		render(<SoundPad />).baseElement;

		// When
		gunJSOnCallback(null);

		// Then
		console.log(gunJSOnCallback);
		expect(gunJSOnCallback).toBeDefined();
		expect(setItemMapMock).toBeCalledTimes(0);
	});
});