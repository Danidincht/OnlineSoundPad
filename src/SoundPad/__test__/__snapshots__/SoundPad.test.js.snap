// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders basic Sound Pad layout 1`] = `
<body>
  <div>
    <div>
      Sound Pad Online
      <div
        class="PadEditor"
      >
        <form>
          <input
            name="title"
            placeholder="Pad title"
            type="text"
            value=""
          />
        </form>
        PadEditor Component
      </div>
    </div>
  </div>
</body>
`;

exports[`renders items from GunJS 1`] = `
<body>
  <div>
    <div>
      Sound Pad Online
      <div
        class="PadEditor"
      >
        <form>
          <input
            name="title"
            placeholder="Pad title"
            type="text"
            value="[object Map]"
          />
        </form>
        PadEditor Component
      </div>
      <div>
        text1
        <audio
          controls=""
          src="[object Object]"
        />
      </div>
      <div>
        text2
        <audio
          controls=""
          src="[object Object]"
        />
      </div>
    </div>
  </div>
</body>
`;
